src/OgHelper.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og\OgHelper.
+ */
+
+namespace Drupal\og;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Field\FieldException;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+
+/**
+ * OG helper methods.
+ */
+class OgHelper {
+
+  /**
+   * Return TRUE if a field can be used and has not reached maximum values.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The content entity to check the field cardinality for.
+   * @param string $field_name
+   *   The field name to check the cardinality of.
+   *
+   * @return bool
+   */
+  public static function checkFieldCardinality(ContentEntityInterface $entity, $field_name) {
```

Comment:
I'm still trying to think what would be better `EntityInterface` or `ContentEntityInterface`. For example in Og::userAccess() we are checking `EntityInterface` and if not a `ContentEntityInterface` we return early.

I usually tend to prefer that API functions would be strict, and throw exceptions if they don't get the right data.

So the question is - should we change other places to use `ContentEntityInterface` as-well?